### ISSUE TITLE
[New Rule] `dict` use get with default rather than `if in mydict: else`

### ISSUE THREAD

**Issue by jcpunk** [MartinThoma/flake8-simplify#72]
## Explanation

Dictionaries have built in methods for "fetch this value if it exists or use this default".  The pattern of checking for the key and then assigning tends to pop up here and there.  It is a good practice to validate the data exists before accessing it, but in this case it is more complex than just using the language buit-in behavior.

## Example

```python
# Bad
if "my_key" in example_dict:
    thing = example_dict["my_key"]
else:
    thing = "default_value"

# Good
thing = example_dict.get("my_key", "default_value")
```

**Comment by MartinThoma** [comment]
That's a good one! I'm uncertain when I will find the time to add it, but I definitely want to add it!

**Comment by MartinThoma** [comment]
```
$ astpretty --no-show-offsets /dev/stdin <<< `cat example.txt`
Module(
    body=[
        If(
            test=Compare(
                left=Constant(value='my_key', kind=None),
                ops=[In()],
                comparators=[Name(id='example_dict', ctx=Load())],
            ),
            body=[
                Assign(
                    targets=[Name(id='thing', ctx=Store())],
                    value=Subscript(
                        value=Name(id='example_dict', ctx=Load()),
                        slice=Constant(value='my_key', kind=None),
                        ctx=Load(),
                    ),
                    type_comment=None,
                ),
            ],
            orelse=[
                Assign(
                    targets=[Name(id='thing', ctx=Store())],
                    value=Constant(value='default_value', kind=None),
                    type_comment=None,
                ),
            ],
        ),
        Assign(
            targets=[Name(id='thing', ctx=Store())],
            value=Call(
                func=Attribute(
                    value=Name(id='example_dict', ctx=Load()),
                    attr='get',
                    ctx=Load(),
                ),
                args=[
                    Constant(value='my_key', kind=None),
                    Constant(value='default_value', kind=None),
                ],
                keywords=[],
            ),
            type_comment=None,
        ),
    ],
    type_ignores=[],
)
```
